atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/config/ConnectionConfig.java

```diff
@@ -168,9 +168,10 @@ public HikariConfig getHikariConfig() {
         // ConnectionConfig.connectionTimeoutSeconds is passed in via getHikariProperties(), in subclasses.
         config.setConnectionTimeout(getCheckoutTimeout());
 
-        // TODO (bullman): See if driver supports JDBC4 (isValid()) and use it.
         config.setConnectionTestQuery(getTestQuery());
```

Comment:
I think we only want to set the connection test query on the hikari config if we have a non-empty test query. That way, if test query is empty, hikari will use the JDBC 4 `isValid()` check internally. See https://github.com/brettwooldridge/HikariCP#frequently-used which states

> 🔤connectionTestQuery
> If your driver supports JDBC4 we strongly recommend not setting this property. This is for "legacy" drivers that do not support the JDBC4 Connection.isValid() API. This is the query that will be executed just before a connection is given to you from the pool to validate that the connection to the database is still alive. Again, try running the pool without this property, HikariCP will log an error if your driver is not JDBC4 compliant to let you know. Default: none

```suggestion
        if (!getTestQuery().isEmpty()) {
            config.setConnectionTestQuery(getTestQuery());
        }
```